src/utils/deep-equal.ts

```diff
@@ -0,0 +1,32 @@
+// This could be done with something like lodash's deepEqual, but in order
```

Comment:
This was previously under `src/validations/` but I don't see the point of having a separate directory just for this. Arguably these should be under `scripts/` and not `src/` anyway, but maybe not.